tests/foreman/ui_airgun/test_remoteexecution.py

```diff
@@ -0,0 +1,242 @@
+"""Test class for Remote Execution Management UI
+
+:Requirement: Remoteexecution
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: UI
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+import datetime
+
+from nailgun import entities
+
+from robottelo.cli.host import Host
+from robottelo.constants import DISTRO_DEFAULT
+from robottelo.datafactory import gen_string
+from robottelo.decorators import fixture, tier3, upgrade
+from robottelo.helpers import add_remote_execution_ssh_key
+from robottelo.vm import VirtualMachine
+
+
+def _setup_vm_client_host(vm_client, org_label, subnet_id=None, by_ip=True):
+    """Setup a VM host for remote execution.
+
+    :param VirtualMachine vm_client: where vm_client is VirtualMachine instance.
+    :param str org_label: The organization label.
+    :param int subnet: (Optioanl) Nailgun subnet entity id, to be used by the vm_client host.
+    :param bool by_ip: Whether remote execution will be affected by ip.
+    """
+    vm_client.install_katello_ca()
+    vm_client.register_contenthost(org_label, lce='Library')
+    assert vm_client.subscribed
+    add_remote_execution_ssh_key(vm_client.ip_addr)
+    if subnet_id is not None:
+        Host.update({
+            'name': vm_client.hostname,
+            'subnet-id': subnet_id,
+        })
+    if by_ip:
+        # connect to host by ip
+        Host.set_parameter({
+            'host': vm_client.hostname,
+            'name': 'remote_execution_connect_by_ip',
+            'value': 'True',
+        })
```

Comment:
Do not think that we should check here, as set parameter  must have it's own tests, we use it here in  a helper function, in general if the command generally fail, the cmd will fail with CLI Error, but if the CLI pass and that parameter is not set, this definitely a bug, any way we can assert on that host parameter value but it's not the responsibility of this function, otherwise we have to ensure everything in every place.

Reply:
ok, fair enough